pkg/network/node/ovscontroller.go

```diff
@@ -161,7 +161,10 @@ func (oc *ovsController) SetupOVS(clusterNetworkCIDR []string, serviceNetworkCID
 	}
 
 	// Link-local traffic
-	otx.AddFlow("table=30, priority=75, ip, nw_dst=169.254.0.0/16, actions=drop")
+	otx.AddFlow("table=30, priority=75, ip, tcp, tcp_dst=80, nw_dst=169.254.0.0/16, actions=drop")
+	otx.AddFlow("table=30, priority=75, ip, tcp, tcp_dst=443, nw_dst=169.254.0.0/16, actions=drop")
+	otx.AddFlow("table=30, priority=75, ip, udp, udp_dst=80, nw_dst=169.254.0.0/16, actions=drop")
+	otx.AddFlow("table=30, priority=75, ip, udp, udp_dst=443, nw_dst=169.254.0.0/16, actions=drop")
```

Comment:
Don't need UDP.
And also, at this point, if we're giving up on the idea of blocking all link-local traffic out of the pod network then this should match `nw_dst=169.254.169.254`, not `nw_dst=169.254.0.0/16`

Reply:
Wait, flows are ordered, right? So can we allow packets with a port (src or dst) of 53, and drop all others?

Reply:
Well, you can't just generically "allow"; you have to say specifically what to do with the packet next (in this case, `goto_table:100`). But... then you'd have two sets of `goto_table:100` rules with a `drop` rule between them and that kind of violates the code style? Handwave handwave.

Yes, we could do it.